tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -204,3 +208,122 @@ def check_node_quota_metric(self, config_prop: self.ConfigProp,
         assert len(errors) == 0, (
             f"Test has failed with {len(errors)} distinct errors. "
             f"{errors}, rnd_seed: {self.rnd_seed}")
+
+
+class ThroughputLimitsSnc(ThroughputLimitsSncBase):
+    """
+    A generic class for tests for throughput limiting that works at
+    shard/node/cluster (SNC) levels
+    """
+    def __init__(self, test_ctx: TestContext, *args, **kwargs):
+        super(ThroughputLimitsSnc, self).__init__(test_ctx, *args, **kwargs)
+        # at least 3 partitions needed overall to have 3 partition leaders
+        # at the same node,
+        self.topics = (TopicSpec(partition_count=3), )
+        self.failed_checks = []
+        self.total_checks = 0
+
+    def check_equal_epsilon(self, measured: int, expected: int, epsilon: float,
+                            what: str):
+        error = (measured - expected) / expected
+        measured = int(measured)
+        if abs(error) > epsilon:
+            self.failed_checks.append(
+                f"{what} ({measured}) is not within ±{epsilon*100}% error "
+                f"of the expected value ({expected}), "
+                f"actual error: {float(f'{error*100:.4g}'):+g}%")
+            self.logger.error(
+                f"(FAIL) {what}: {measured} == {expected} {float(f'{error:.4g}'):+g} < ±{epsilon*100}%"
+            )
+        else:
+            self.logger.info(
+                f"(PASS) {what}: {measured} == {expected} {float(f'{error:.3g}'):+g} < ±{epsilon*100}%"
+            )
+        self.total_checks += 1
+
+    @cluster(num_nodes=4)
+    @parametrize(config_quota_node_max=8 * 1024 * 1024)
+    @parametrize(config_quota_node_max=2 * 1024 * 1024)
+    @parametrize(config_quota_node_max=512 * 1024)
+    def test_node_limits(self, config_quota_node_max):
+        """
+        Apply various backpressure to one node of the cluster.
+        Verify that node throughput limits are applied.
+        """
+
+        # add partitions to the topic until there are 3 partition
+        # leaders on any of the nodes
+        partitions_cnt = len(self.redpanda.partitions(self.topic))
+        while True:
+            while True:
```

Comment:
Maybe it would be nice to bound these loops with a max_timeout or max number of retries